### DISCUSSION TITLE
New get param method support for Request class: get_param_as_enum

### DISCUSSION THREAD

**Discussion by DeloitteYuvarajan** [falconry/falcon#2093] [Ideas]
https://github.com/falconry/falcon/blob/492087f8f9e0363f452af94fc919622b4866ec78/falcon/request.py#L49

Suggested:

```
def get_param_as_enum(self, name, enum_class, required=False, store=None, default=None):
    if value := self.get_param(name, required=required):
        try:
            value = enum_class[value]
            return value
        except KeyError:
            try:
                value = enum_class(value)
                return value
            except ValueError:
                raise HTTPInvalidParam(
                    title='Invalid', description=f'Invalid param {value}')
    if not required:
        return default
```
PS: I might miss store functionality, we can add it later.

**Comment by CaselIT** [comment]
Hi,

That would be fine by me, but I wonder if we could create an helper method that handles the common logic?

Cc @vytas7

**Comment by vytas7** [comment]
Hm dunno, let's maybe discuss pros and cons first?

This can be easily implemented by the user if needed, and it is probably better to use a validation library for advanced use cases rather than trying to implement every possible validator via `get_param_as_*()`.
